manage.py

```diff
@@ -311,6 +313,27 @@ def update_node_packages(project_config):
     except Exception as e:
         print("An error occurred during Node.js package update: " + str(e))
 
+def toggle_maintenance_mode(project_config):
+    project_path = project_config['path']
+    deploying_user = project_config['deploying_user']
+    artisan_path = os.path.join(project_path, 'artisan')
+    maintenance_file = os.path.join(project_path, 'storage', 'framework', 'maintenance.php')
+
+    if not os.path.exists(artisan_path):
+        print(f"Error: Maintenance not supported for this project. Operation aborted.")
+        sys.exit(1)
+
+    action = 'up' if os.path.exists(maintenance_file) else 'down'
+
+    command = f"sudo -u {deploying_user} php {artisan_path} {action}"
```

Comment:
Should this be done in the container to avoid php compat issues?

Reply:
Yes, actually. Ugh it’s been a day